core/trie/storage.go

```diff
@@ -79,10 +85,18 @@ func (t *Storage) Get(key *BitArray) (*Node, error) {
 	}
 
 	var node *Node
-	err = t.txn.Get(buffer.Bytes(), func(val []byte) error {
-		node = nodePool.Get().(*Node)
-		return node.UnmarshalBinary(val)
-	})
+
+	if t.snapshot != nil {
+		err = t.snapshot.Get(buffer.Bytes(), func(val []byte) error {
+			node = nodePool.Get().(*Node)
+			return node.UnmarshalBinary(val)
+		})
+	} else {
+		err = t.txn.Get(buffer.Bytes(), func(val []byte) error {
+			node = nodePool.Get().(*Node)
+			return node.UnmarshalBinary(val)
+		})
+	}
```

Comment:
Let me know if I misunderstand something here, but this means that if `snapshot` is configured, we don't read from `txn`. This means any write done in `txn` can't be retrieved, which means `PendingState` will be incorrect.